pac4j-cas/src/main/java/org/pac4j/cas/client/CasClient.java

```diff
@@ -266,7 +266,7 @@ protected void initializeClientConfiguration() {
             this.casPrefixUrl += "/";
         }
         if (CommonHelper.isBlank(this.casPrefixUrl)) {
-            this.casPrefixUrl = this.casLoginUrl.replaceFirst("/login", "/");
+            this.casPrefixUrl = this.casLoginUrl.replaceFirst("/login$", "/$");
```

Comment:
:-1: on this. Nobody will remember to do this. We'll need to be smarter about replacing bits in the path. 

Also, this only seems to be a problem if you don't set the casPrefix url? Can we simply set that value and move on?